python/pyarrow/table.pxi

```diff
@@ -2442,6 +2602,46 @@ def _from_pydict(cls, mapping, schema, metadata):
         raise TypeError('Schema must be an instance of pyarrow.Schema')
 
 
+def _from_pylist(cls, mapping, schema, metadata):
+    """
+    Construct a Table/RecordBatch from list of dictionary of rows.
+
+    Parameters
+    ----------
+    cls : Class Table/RecordBatch
+    mapping : list of dicts of rows
+        A mapping of strings to row values.
+    schema : Schema, default None
+        If not passed, will be inferred from the Mapping values.
```

Comment:
Maybe you can specify here that it will be inferred from the _first_ row (also in the actual user facing docstrings above)